File: bundle/BarcodeReaderCLI/examples/brcli-example.rb
# encoding: utf-8
#  ruby brcli-example.rb           ->  use BarcodeReaderCLI command line 
#  ruby brcli-example.rb config    ->  use BarcodeReaderCLI configuration file  

require 'open3'
dir = File.dirname(__FILE__)  # enable 'require' use modules in script folder
Dir.chdir(dir)

args  = Array[
'../bin/BarcodeReaderCLI',
'-type=code128',
'"https://wabr.inliteresearch.com/SampleImages/1d.pdf"',
]

configFile = './brcli-example.config'
argsConfig  = Array[
'../bin/BarcodeReaderCLI',
'@"' + configFile + '"'
]

args = argsConfig  if ARGV.length > 0
params = args.join(' ')
output, error, status = Open3.capture3(params)

if output != "" then  
  puts "STDOUT:\n " + output
end
if error != "" then 
   puts "STDERR:\n " + error
end

File: bundle/BarcodeReaderCLI/examples/brcli-example.php
<?php
// php brcli-example.php          ->  use BarcodeReaderCLI command line 
// php brcli-example.php config   ->  use BarcodeReaderCLI configuration file 
chdir(__DIR__);

$args = array(
'"../bin/BarcodeReaderCLI"',
'-type=code128',
'"https://wabr.inliteresearch.com/SampleImages/1d.pdf"',
);

$configFile = './brcli-example.config';
$argsConfig = array(
'"../bin/BarcodeReaderCLI"',
'@"' . $configFile . '"',
);

function doExec($cmd, &$stdout=null, &$stderr=null) {
    $proc = proc_open($cmd,[
        1 => ['pipe','w'],
        2 => ['pipe','w'],
    ],$pipes);
    $stdout = stream_get_contents($pipes[1]);
    fclose($pipes[1]);
    $stderr = stream_get_contents($pipes[2]);
    fclose($pipes[2]);
    return proc_close($proc);
}

$output = ""; $error = "";
$params = implode(" ", $argc > 1 ?  $argsConfig : $args);
if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') $params = '"' . $params . '"'; 

doExec($params, $output, $error);
if ($output != "") echo "STDOUT:\n $output";
if ($error != "") echo "STDERR:\n $error";
?>